docs/current_docs/api/constructors.mdx

```diff
@@ -126,12 +117,26 @@ Here is an example of a Dagger module with a default constructor argument of typ
 ```python file=./snippets/constructors/python/default-object/main.py
 ```
 
+:::info
+You can use other [`@dataclasses.dataclass`](https://docs.python.org/3/library/dataclasses.html) decorators to generate more complex constructors. For example:
+
+```python file=./snippets/constructors/python/complex/main.py
+```
+
+A few important notes:
+
+- Class attributes become constructor arguments by default, unless specified as `field(init=False)`.
+- The [`__post_init__` function](https://docs.python.org/3/library/dataclasses.html#post-init-processing) is called at the end of the auto-generated constructor. So by setting `init=False` on the container argument, it changes from being a constructor argument to a regular argument that is still required.
+- Setting a default value like `None` in an attribute is the same as setting `@dataclasses.field(default=None)`. `@dataclasses.field()` is only required to specify other non-default values, like `init` or to set the default using a [factory function](https://docs.python.org/3/library/dataclasses.html#default-factory-functions).
```

Comment:
Not decorators:
```suggestion
- Setting a default value like `None` in an attribute is the same as setting `dataclasses.field(default=None)`. `dataclasses.field()` is only required to specify other non-default values, like `init` or to set the default using a [factory function](https://docs.python.org/3/library/dataclasses.html#default-factory-functions).
```
This is standard dataclass. You could also need `dagger.field()` if you want it exposed as a function in the API, with a default value or not.